galaxy_ng/app/utils/galaxy.py

```diff
@@ -0,0 +1,111 @@
+import logging
+import requests
+from urllib.parse import urlparse
+
+
+logger = logging.getLogger(__name__)
+
+
+def paginated_results(next_url):
+
+    parsed = urlparse(next_url)
+    _baseurl = parsed.scheme + '://' + parsed.netloc
+    results = []
+    while next_url:
+        logger.info(f'fetch {next_url}')
+        rr = requests.get(next_url)
+        ds = rr.json()
+        results.extend(ds['results'])
+        if ds['next_link']:
+            next_url = _baseurl + ds['next_link']
+        else:
+            next_url = None
+    return results
+
+
+def upstream_role_iterator(
+    baseurl=None,
+    limit=None,
+    github_user=None,
+    github_repo=None,
```

Comment:
Doesn't seem `github_repo` is ever passed in, but maybe that is expected for future edits

Reply:
These will allow for future filtering of which roles to synchronize. For example we might want to sync everything from a specific github org/user or a github repo.